taskcluster/xpi_taskgraph/transforms/signing.py

```diff
@@ -19,61 +19,75 @@
 
 @transforms.add
 def define_signing_flags(config, tasks):
-    for task in tasks:
-        dep = task["primary-dependency"]
-        # Current kind will be prepended later in the transform chain.
-        task["name"] = _get_dependent_job_name_without_its_kind(dep)
-        attributes = dep.attributes.copy()
-        if task.get("attributes"):
-            attributes.update(task["attributes"])
-        task["attributes"] = attributes
-        task["attributes"]["signed"] = True
-        if "run_on_tasks_for" in task["attributes"]:
-            task.setdefault("run-on-tasks-for", task["attributes"]["run_on_tasks_for"])
+    if (
+        config.params.get("version")
+        and config.params.get("xpi_name")
+        and config.params.get("head_ref")
+        and config.params.get("build_number")
+        and config.params.get("level")
+    ):
+        for task in tasks:
+            dep = task["primary-dependency"]
+            # Current kind will be prepended later in the transform chain.
+            task["name"] = _get_dependent_job_name_without_its_kind(dep)
+            attributes = dep.attributes.copy()
+            if task.get("attributes"):
+                attributes.update(task["attributes"])
+            task["attributes"] = attributes
+            task["attributes"]["signed"] = True
+            if "run_on_tasks_for" in task["attributes"]:
+                task.setdefault("run-on-tasks-for", task["attributes"]["run_on_tasks_for"])
 
-        for key in ("worker-type", "worker.signing-type"):
-            resolve_keyed_by(
-                task, key, item_name=task["name"], level=config.params["level"]
-            )
-        yield task
+            for key in ("worker-type", "worker.signing-type"):
+                resolve_keyed_by(
+                    task, key, item_name=task["name"], level=config.params["level"]
+                )
+            yield task
 
 
 @transforms.add
 def build_signing_task(config, tasks):
-    for task in tasks:
-        dep = task["primary-dependency"]
-        task["dependencies"] = {"build": dep.label}
-        if not dep.task["payload"]["env"]["ARTIFACT_PREFIX"].startswith("public"):
-            scopes = task.setdefault("scopes", [])
-            scopes.append(
-                "queue:get-artifact:{}/*".format(
-                    dep.task["payload"]["env"]["ARTIFACT_PREFIX"].rstrip("/")
+    if (
+        config.params.get("version")
+        and config.params.get("xpi_name")
+        and config.params.get("head_ref")
+        and config.params.get("build_number")
+        and config.params.get("level")
+    ):
```

Comment:
Hm. I *think* this block may have resulted in no longer running depsigning tasks anywhere?

I:

```bash
# generate all the graphs on main
../taskgraph-diff/taskgraph-gen.py --params-dir ../taskgraph-diff/params-xpi --full xpi --overwrite 2>&1 | tee out
# generate all the graphs on add-balrog-release-task
git checkout add-balrog-release-task
../taskgraph-diff/taskgraph-gen.py --params-dir ../taskgraph-diff/params-xpi --full gabriel --overwrite 2>&1 | tee out
# diff the two sets of graphs
../taskgraph-diff/taskgraph-diff.py xpi gabriel
```

Now in `../taskgraph-diff/json/gabriel/*.diff` I notice dep-signing only appears where we have a xpi_name specified, where we used to have it in pull requests and on push.

We may want to amend this to:

```python
    if (
        config.kind == "release-signing" and not (
            config.params.get("version")
            and config.params.get("xpi_name")
            and config.params.get("head_ref")
            and config.params.get("build_number")
            and config.params.get("level")
        )
    ):
        continue
```

if I haven't made a mental error. But you have this block at least twice. Since we seem to do this in multiple transform functions, maybe let's make this its own transform function? Before any other transforms in this file, we could

```python
@transforms.add
def restrict_release_signing_to_xpi_name(config, tasks):
    for task in tasks:
        if (
            config.kind != "release-signing" or (
                config.params.get("version")
                and config.params.get("xpi_name")
                and config.params.get("head_ref")
                and config.params.get("build_number")
                and config.params.get("level")
            )
        ):
            yield task
```

which will limit `tasks` to just dep-signing tasks or release-signing for a given release, and we can just proceed with simpler transforms later.